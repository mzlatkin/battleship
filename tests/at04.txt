-- student test 4
custom_setup_test(4,2,8,3)
fire([A,1])
give_up
custom_setup_test(4,2,8,3)
