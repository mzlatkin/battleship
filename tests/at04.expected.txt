  state 0 OK -> Start a new game
->custom_setup_test(4,2,8,3)
  state 1 OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/3
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->fire([A, 1])
  state 2 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 0/3
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->give_up
  state 3 OK -> You gave up!
     1  2  3  4
  A  O  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 0/3
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->custom_setup_test(4,2,8,3)
  state 4 OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  _
  C  _  v  _  _
  D  _  v  _  v
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/3
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [C, 2]->v;[D, 2]->v
    1x1: [D, 4]->v
